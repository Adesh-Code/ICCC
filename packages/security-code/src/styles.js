const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
