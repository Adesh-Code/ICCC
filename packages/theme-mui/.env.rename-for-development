# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE