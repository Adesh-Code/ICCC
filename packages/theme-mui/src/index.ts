// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
