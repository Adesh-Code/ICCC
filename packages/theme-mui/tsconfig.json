{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
