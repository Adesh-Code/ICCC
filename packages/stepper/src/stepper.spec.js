import React from "react";
import { createTheme, ThemeProvider } from "@mui/material/styles";
import { render } from "@testing-library/react";
import { StepIconComponent } from ".";
import { themeLight } from "@iccc/theme-mui";
import "@testing-library/jest-dom";

const light = createTheme(themeLight);

const arrange = ({ ...props }) =>
  render(
    <ThemeProvider theme={light}>
      <StepIconComponent {...props} />
    </ThemeProvider>
  );

describe("Step icon component", () => {
  it("renders an error icon", () => {
    const { container } = arrange({ error: true });
    expect(container).toMatchSnapshot();
  });
  it("renders an error icon even if other props are provided", () => {
    const { container } = arrange({ completed: true, error: true });
    expect(container).toMatchSnapshot();
  });
  it("renders a completed icon", () => {
    const { container } = arrange({ completed: true });
    expect(container).toMatchSnapshot();
  });
  it("renders an incomplete icon", () => {
    const { container } = arrange({ icon: null });
    expect(container).toMatchSnapshot();
  });
  it("renders a numeric incomplete icon", () => {
    const { container } = arrange({ icon: 1 });
    expect(container).toMatchSnapshot();
  });
  it("renders an active incomplete icon", () => {
    const { container } = arrange({ active: true });
    expect(container).toMatchSnapshot();
  });
  it("renders an active numeric incomplete icon", () => {
    const { container } = arrange({ active: true, icon: 1 });
    expect(container).toMatchSnapshot();
  });
});
