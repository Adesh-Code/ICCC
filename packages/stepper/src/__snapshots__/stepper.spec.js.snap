// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step icon component renders a completed icon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-0"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg
      fill="none"
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M21.6538 11.3518C22.0731 11.7113 22.1217 12.3426 21.7622 12.7619L15.4192 20.1621C15.2413 20.3696 14.9863 20.4953 14.7133 20.5099C14.4404 20.5245 14.1734 20.4268 13.9744 20.2394L10.7589 17.2121C10.3567 16.8335 10.3377 16.2006 10.7162 15.7985C11.0948 15.3964 11.7277 15.3773 12.1298 15.7559L14.5825 18.065L20.2437 11.4603C20.6031 11.041 21.2344 10.9924 21.6538 11.3518Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </svg>
</div>
`;

exports[`Step icon component renders a numeric incomplete icon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-0"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <text
      dominant-baseline="central"
      text-anchor="middle"
      x="12"
      y="12"
    >
      1
    </text>
  </svg>
</div>
`;

exports[`Step icon component renders an active incomplete icon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-0"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg
      fill="none"
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="16"
        cy="16"
        fill="currentColor"
        r="4"
      />
    </svg>
  </svg>
</div>
`;

exports[`Step icon component renders an active numeric incomplete icon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-0"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <text
      dominant-baseline="central"
      text-anchor="middle"
      x="12"
      y="12"
    >
      1
    </text>
  </svg>
</div>
`;

exports[`Step icon component renders an error icon 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-0"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg
      fill="none"
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M16 9.66669C16.5523 9.66669 17 10.1144 17 10.6667V17.3334C17 17.8856 16.5523 18.3334 16 18.3334C15.4477 18.3334 15 17.8856 15 17.3334V10.6667C15 10.1144 15.4477 9.66669 16 9.66669ZM16 21C16.5523 21 17 21.4477 17 22V22.1481C17 22.7004 16.5523 23.1481 16 23.1481C15.4477 23.1481 15 22.7004 15 22.1481V22C15 21.4477 15.4477 21 16 21Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </svg>
</div>
`;

exports[`Step icon component renders an error icon even if other props are provided 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-0"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <svg
      fill="none"
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M16 9.66669C16.5523 9.66669 17 10.1144 17 10.6667V17.3334C17 17.8856 16.5523 18.3334 16 18.3334C15.4477 18.3334 15 17.8856 15 17.3334V10.6667C15 10.1144 15.4477 9.66669 16 9.66669ZM16 21C16.5523 21 17 21.4477 17 22V22.1481C17 22.7004 16.5523 23.1481 16 23.1481C15.4477 23.1481 15 22.7004 15 22.1481V22C15 21.4477 15.4477 21 16 21Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </svg>
</div>
`;

exports[`Step icon component renders an incomplete icon 1`] = `<div />`;
